### DISCUSSION TITLE
Is it possible to pass vector parameters using equation_systems.parameters.set

### DISCUSSION THREAD

**Discussion by nuomi68** [libMesh/libmesh#3724] [Q&A] [answered]
Is it possible to pass vector parameters using equation_systems.parameters.set, and if so, how? Are there any examples?
What does the get_solution (std::vector<Number> & soln,std::vector<std::string> & names) function do? How does it work? Are there examples?
thank youstore_initial_solution(

**Comment by jwpeterson** [comment, marked as answer]
> Is it possible to pass vector parameters using equation_systems.parameters.set, and if so, how? 

I don't think it's possible. If you want to pass more than scalar integer or floating point values to your Assembly routine, you should consider using the `System::attach_assemble_object()` API instead of `System::attach_assemble_function()`. The former allows you to define a C++ class with any contents you like, while the latter requires you to store data on `EquationSystems` object via the `set()` API and is less flexible. See systems_of_equations_ex6 and ex9 for examples of using `System::attach_assemble_object()`.


> What does the get_solution (std::vector & soln,std::vectorstd::string & names) function do?

`get_solution()` is a way to get just the "elemental" components of the full solution vector, and was used mainly for doing I/O if I recall correctly.  This function is deprecated, so you likely won't find it being called anywhere, and should not call it yourself.

```
  /**
   * Retrieve the solution data for CONSTANT MONOMIALs and/or components of
   * CONSTANT MONOMIAL_VECs. If 'names' is populated, only the variables
   * corresponding to those names will be retrieved. This can be used to
   * filter which variables are retrieved.
   *
   * \deprecated Call the more appropriately-named build_elemental_solution_vector()
   * instead.
   */
  void get_solution (std::vector<Number> & soln,
                     std::vector<std::string> & names) const;
```

**Reply by nuomi68** [reply]
Thank you. I will try further！

**Comment by roystgnr** [comment]
We do have `eq_sys.parameters.set<std::vector<dof_id_type> >("charged_elem_id")=charged_elem_ids;` in `miscellaneous_ex15.C`, and a glance at `parameters.h` says we support vectors-of-vectors and `std::map` as parameters too.  But I do think John's right that using your own assembly object class (or even your own `System` subclass) is a better alternative here.

**Reply by nuomi68** [reply]
Thanks for the information, I will try to follow this example！
